src/data/viteComboFetcher.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * **WARNING** This module is expected to be invoked from a Vite Plugin!
+ * 
+ * Aggregated all YAML Files in the ./combos Directory and interpretes 
+ */
+
+import { parse } from "yaml"
+import { DbCombosTableZod, DbTricksTableZod } from "../lib/database/schemas/CurrentVersionSchema"
+import { readFile } from "node:fs/promises"
+import { ZodError, z } from "zod"
+import {globby} from "globby"
+import { join } from "node:path"
+import { fileURLToPath } from "node:url"
+import { YamlComboTableSchemaZod } from "./viteSchemaGenerator"
+
+async function fetchYamlFile(path: string) {
+  const asObject = YamlComboTableSchemaZod.parse(parse(await readFile(path, "utf8")))
+
+  if(!asObject.tricks) {
+    throw new Error("Combos must contain at least one trick.")
+  }
+
+  asObject.tricks = asObject.tricks.map( trickId => ([trickId, "official"])) as any
+
+  // Makes sure the YAML has the right structure. 
+  return DbCombosTableZod.parse({...asObject, comboStatus: "official"})
+}
+
+/**
+ * This takes a list of Database Entities and creates a mapping of ID → List of Objects
+ * 
+ * This can be used to detect duplicates or to check if an Entity with a given ID exists.
+ */
+function createRecordLookup<T extends z.infer<typeof DbCombosTableZod> | z.infer<typeof DbTricksTableZod>>(allEntities: T[]) {
+  let idToEntityLookup: Record<number, T[]> = {};
+
+  allEntities.forEach(databaseEntity => {
+    if (!idToEntityLookup[databaseEntity.id]) {
+      idToEntityLookup[databaseEntity.id] = [];
+    }
+      idToEntityLookup[databaseEntity.id].push(databaseEntity);
+    });
+  return idToEntityLookup;
+}
+
+function findDuplicateKeys(allCombos: z.infer<typeof DbCombosTableZod>[]) {
+
+  let idToTrickLookup: Record<number, (z.infer<typeof DbCombosTableZod>)[]> = createRecordLookup(allCombos);
+
+  return Object.entries(idToTrickLookup).filter(([_,trick]) => trick.length > 1).map(([k,v]) => ({
```

Comment:
I guess `trick` should be something like `tricksInCombo`?

Reply:
Also when I hover over the trick I get the following information:
<img width="1092" alt="Bildschirmfoto 2023-09-15 um 12 26 52" src="https://github.com/bastislack/highline-freestyle/assets/74077179/38d58ad3-e96c-4f35-bd2e-807599c739e6">

Why is there a comboStatus?
Maybe `([_,trick(sInCombo)])` is actually `([_,combo])`?

Reply:
Yep, good shout :)

fyi, `Object.entries` turns the entries of an object into a tuple-Array of `[key,value][]` 

so 
```ts
{
  x: 42
  a: "hello world"
}
```
is turned into 
```ts
[
  ["x", 42],
  ["a", "hello world"], 
]
```
there is also `Object.fromEntries` which will take such a tuple and construct an object from it.